### ISSUE TITLE
Exclude extension pattern is not working

### ISSUE THREAD

**Issue by brimwats** [trganda/obsidian-attachment-management#18]
Hi! I have these settings, and evertytime I switch awy from the settings the exclude extension does not save:

![Obsidian_6s8krd6UUd](https://github.com/trganda/obsidian-attachment-management/assets/73113656/655c31c3-6ebf-4a67-8eea-7d26d4d03e89)

**Comment by brimwats** [comment]
Here's debug if you need it:

```
SYSTEM INFO:
	Obsidian version: v1.3.2
	Installer version: v1.2.8
	Operating system: Windows 10 Pro 10.0.22621
	Login status: logged in
	Catalyst license: insider
	Insider build toggle: on
	Live preview: on
	Legacy editor: off
	Base theme: light
	Community theme: Minimal v6.3.3
	Snippets enabled: 43
	Restricted mode: off
	Plugins installed: 90
	Plugins enabled: 69
		1: Advanced URI v1.35.0
		2: Divide & Conquer v1.0.0
		3: Another Quick Switcher v8.10.0-beta2
		4: Zotero Integration v2.3.9
		5: Wikipedia v1.0.3
		6: Templater v1.16.0
		7: Tag Wrangler v0.5.8
		8: Supercharged Links v0.9.7
		9: Style Settings v1.0.3
		10: Smarter Markdown Hotkeys v1.16.4
		11: Settings Search v1.3.7
		12: Banners v1.3.3
		13: Better Command Palette v0.15.0
		14: Better Word Count v0.9.6
		15: Breadcrumbs v3.6.4
		16: Code Editor Shortcuts v1.13.1
		17: Commander v0.5.0
		18: Contextual Typography v2.2.5
		19: Custom Frames v2.4.6
		20: DB Folder v3.4.0
		21: Dataview v0.5.56
		22: Dynamic Highlights v0.3.1
		23: Dynamic Table of Contents v0.0.27
		24: Excalidraw v1.9.2
		25: File path to URI v1.4.1
		26: Footnote Shortcut v0.1.2
		27: Format Hotkeys v0.1.7
		28: Hider v1.2.4
		29: Map View v3.0.2
		30: Metadata Menu v0.4.21
		31: Note Auto Creator v1.4.0
		32: Obsidian Charts v3.7.1
		33: Obsidian Columns v1.4.1
		34: Obsidian42 - BRAT v0.7.0
		35: Omnisearch v1.14.1-beta.1
		36: Periodic Notes v1.0.0-beta.3
		37: Pane Relief v0.5.0
		38: QuickAdd v1.0.2
		39: Readwise Official v2.0.1
		40: Reveal Active File Button v2.0.3
		41: Sentence Navigator v1.3.0
		42: Keyshots v2.0.1
		43: LanguageTool Integration v0.3.3
		44: Pandoc Plugin v0.4.1
		45: Quick Explorer v0.2.8
		46: floating toc v2.3.0
		47: Hover Editor v0.11.11
		48: Vimrc Support v0.9.0
		49: Jump to link v0.4.6
		50: Vim Yank Highlighter v1.1.4
		51: Plugin Groups v2.1.0
		52: Callout Manager v1.0.1
		53: Global Search and Replace v0.2.0
		54: Awesome Image v0.1.2
		55: Vim Reading View Navigation v1.2.12
		56: Recent Files v1.3.5
		57: Keyboard Analyzer v0.1.3
		58: ChatGPT MD v1.5.0
		59: Text Generator v0.3.18
		60: Hotkey Helper v0.3.17
		61: Auto Classifier v1.1.1
		62: Folder notes v1.2.3
		63: Plugin Update Tracker v1.5.0
		64: Google Calendar v1.9.14
		65: Vim Toggle v1.0.8
		66: Minimal Theme Settings v6.3.1
		67: Notion-Like Tables v6.9.3
		68: Table Enhancer v0.5.10
		69: Attachment Management v0.4.0

RECOMMENDATIONS:
	Custom theme and snippets: for cosmetic issues, please first try updating your theme and disabling your snippets. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community theme and snippets.
	Community plugins: for bugs, please first try updating all your plugins to latest. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community plugins.

```

**Comment by trganda** [comment]
I have forget to set the updated setting on view.

```ts
      .addText((text) =>
        text
          .setPlaceholder("pdf|docx?|xlsx?|pptx?|zip|rar")
          .setValue("") // it's always ""
          .onChange(async (value) => {
            this.plugin.settings.excludeExtensionPattern = value;
            await this.plugin.saveSettings();
          })
      );
```

But the setting already been save and should work.

I will fix this later.

**Comment by trganda** [comment]
fixed in https://github.com/trganda/obsidian-attachment-management/releases/tag/0.4.1

**Comment by brimwats** [comment]
confirmed solved in 4.1
